src/App/Multisite.php

```diff
@@ -135,4 +125,35 @@ public function getClientUrl($host = null)
 
         return $this->subdomain . '.' . getenv('MULTISITE_CLIENT_DOMAIN');
     }
+    protected function checkDeploymentStatus($deployment)
+    {
+        // No deployment? throw a 404
+        if (! count($deployment)) {
+            abort(404, "Deployment not found");
+        }
+
+        $status = $deployment['status'];
+        $deployed_date = $deployment['deployed_date'];
```

Comment:
You could drop these two lines and just use the array access in the conditionals. That should reduce some of the codeclimate's complaints about complexity